vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/ElementSteps.java

```diff
@@ -373,6 +379,41 @@ public void elementHasWidthRelativeToParentElement(int width)
         });
     }
 
+    /**
+     * Checks if the element located by the specified locater is or is not presented in the browser viewport
+     *
+     * @param locator The locator of the element to check presence in viewport
+     * @param presence The presence state of the element, either <b>is</b> or <b>is not</b>
+     */
+    @Then("element located by `$locator` $presence visible in viewport")
+    public void checkElementViewportPresence(Locator locator, ViewportPresence presence)
+    {
+        List<WebElement> elements = baseValidations.assertNumberOfElementsFound("Element to check", locator, 1,
+                ComparisonRule.LESS_THAN_OR_EQUAL_TO);
+        if (elements.size() == 1)
+        {
+            String script = """
+                const rect = arguments[0].getBoundingClientRect();
+                const windowHeight = window.innerHeight;
+
+                return (rect.top >= 0 && rect.top <= windowHeight) || (rect.bottom > 0 && rect.bottom <= windowHeight);
+                """;
+            boolean inViewport = javascriptActions.executeScript(script, elements.get(0));
```

Comment:
in order to avoid copy-pasted logic it makes sense to introduce abstraction, like, `ScrollActions` and `ScrollValidations`